packages/core/util/layouts/GranularRectLayout.ts

```diff
@@ -306,12 +306,15 @@ export default class GranularRectLayout<T> implements BaseLayout<T> {
 
   private maxHeight: number
 
+  private displayMode: string
+
   private pTotalHeight: number
 
   constructor(args: {
     pitchX: number | undefined
     pitchY: number | undefined
     maxHeight: number | undefined
+    displayMode: string | undefined
```

Comment:
is there some way to add a more restrictive type here that restricts the displayMode to undefined, 'normal', or 'collapsed'?

Reply:
typescript can handle this type of thing, ya. I guess it just depends on, for some types of things like this, whether you get benefit from the type system by being that granular since of course it's not really a run time thing